core/base/src/TROOT.cxx

```diff
@@ -2797,9 +2798,11 @@ void TROOT::SetWebDisplay(const char *webdisplay)
    }
 
    if (fIsWebDisplay) {
+      gEnv->SetValue("Canvas.Name", canName);
```

Comment:
Why do you set `"Canvas.Name"` to the exact value that you got from `gEnv->GetValue("Canvas.Name", "");` above?

Reply:
If `gROOT->SetWebDisplay()` called several times like enable/disable/enable, then at some point one may want to restore default value, which is stored in `.rootrc` file.

Reply:
Thanks, makes sense! Could you add this as a comment, explaining why you seem to do a no-op which in fact is not a no-op for repeated calls?